# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.cpp
# Commit: ee72c01626bf
# Full Hash: ee72c01626bf10d81fcef9e7ed3e1cf8fb00322e
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2024-01-23 05:36:48
# Description:
#   Bug 1875407 downgrade MediaTrackGraph shutdown diagnostic assertion to MOZ_ASSERT() r=padenot
#   
#   We have plenty of crashes from Nightly users to investigate.
#   Leaving the MOZ_ASSERT() in place will catch any instances in CI.
#   
# ==============================================================================

diff -r fed5165a02c0 -r ee72c01626bf dom/media/MediaTrackGraph.cpp
--- a/dom/media/MediaTrackGraph.cpp	Mon Jan 22 19:09:12 2024 +0000
+++ b/dom/media/MediaTrackGraph.cpp	Mon Jan 22 19:11:14 2024 +0000
@@ -1712,8 +1712,7 @@
 NS_IMETHODIMP
 MediaTrackGraphImpl::Notify(nsITimer* aTimer) {
   MOZ_ASSERT(NS_IsMainThread());
-  MOZ_DIAGNOSTIC_ASSERT(!mShutdownBlocker,
-                        "MediaTrackGraph took too long to shut down!");
+  MOZ_ASSERT(!mShutdownBlocker, "MediaTrackGraph took too long to shut down!");
   // Sigh, graph took too long to shut down.  Stop blocking system
   // shutdown and hope all is well.
   RemoveShutdownBlocker();
